### PR TITLE
Live export hardening: normalize providers and validate query mapping robustly

### PR DESCRIPTION
**Pull request by robertbartlomiejski**
## Summary
This PR hardens live research export input handling to fail clearly on invalid query files and consistently normalize provider names.

## What Changed
- `scripts/export_live_research_records.py`
  - Accepts `yaml.safe_load(...) == None` as empty mapping path and emits clear error when no valid query groups exist.
  - Distinguishes invalid YAML mapping from missing/empty `query_groups`.
  - Preserves strict validation messaging for operators.
- `tests/test_export_live_research_records.py`
  - Updated expected error messages for new validation paths.
  - Added provider normalization assertions (single provider and comma-separated with spacing/case).

## Why
- Prevent silent/unclear failures for empty/comment-only query files.
- Ensure provider inputs like `Crossref` and ` Crossref, Scopus ` are normalized before registry search.

## Validation Checklist
- [x] Script behavior updated for empty/comment-only and scalar YAML edge cases
- [x] Test expectations aligned with new error messaging
- [x] Added normalization test for comma-separated providers
- [x] `python -m pytest tests/test_export_live_research_records.py -q` passed locally

## Risk
- Low/medium: controlled input-validation behavior changes plus accompanying tests.

### CODE REVIEW

**Comment by robertbartlomiejski** [comment]
Reviewer notes:
- Please focus on input-hardening behavior in `scripts/export_live_research_records.py` and aligned assertions in `tests/test_export_live_research_records.py`.
- Key behaviors:
  1) clear error for non-mapping YAML,
  2) clear error for missing/empty `query_groups`,
  3) provider normalization for case/spacing and comma-separated input.
- Local validation run: `python -m pytest tests/test_export_live_research_records.py -q`.
- Scope excludes unrelated workflow/domain logic.

**Comment by robertbartlomiejski** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/robertbartlomiejski/morskamary/pull/156#pullrequestreview-4279523726)

File: scripts/export_live_research_records.py [hunk 1]
```diff
@@ -341,18 +341,25 @@ def main() -> int:
...
+        query_config: Dict[str, Any] = {}
+    elif isinstance(query_config_raw, dict):
+        query_config = query_config_raw
+    else:
         print(
-            f"Error: Query file is empty or not a valid YAML mapping: {query_file_path}",
+            f"Error: Query file is not a valid YAML mapping: {query_file_path}",
             file=sys.stderr,
         )
         return 1
```
[the remark this answers is not in the document]
Comment (robertbartlomiejski) [reply scripts/export_live_research_records.py]: @copilot apply changes based on the comments also considering to add into consideration below comment of mine.

To fix this gracefully and maintain your established CLI error-handling patterns, we need to wrap the `yaml.safe_load()` call in a `try/except` block and introduce a new test. 

### 1. Code Implementation to apply
In `scripts/export_live_research_records.py` (around line 268 [1]), the logic should be updated to look like this:

```python
    try:
        query_config = yaml.safe_load(f)
    except yaml.YAMLError as exc:
        print(f"Error: Failed to parse '{path}'. Syntactically invalid YAML:\n{exc}", file=sys.stderr)
        sys.exit(1)
        
    if not isinstance(query_config, dict):
        print(f"Error: '{path}' is empty or not a valid YAML mapping.", file=sys.stderr)
        sys.exit(1)
```
***

**Task Title:** Handle YAML Parse Errors Gracefully in Live Export Script
**Target Repository:** `robertbartlomiejski/morskamary`
**Executive Summary:** `yaml.safe_load()` currently bubbles up a raw `yaml.YAMLError` traceback if the query configuration file is syntactically invalid. Wrap this in a `try/except` block to catch the error, print a clear user-facing message containing the file path, and exit with code 1. This must remain strictly distinct from the structural type validation (`isinstance(query_config, dict)`) implemented previously.

**Step 1: Refactor `scripts/export_live_research_records.py`**
*   **Action:** Locate the `yaml.safe_load(f)` call. 
*   **Constraint:** Wrap it in a `try/except yaml.YAMLError as exc:` block. If caught, print a clear error to `sys.stderr` that includes the file path and the syntax exception details. Call `sys.exit(1)`. Ensure the `import yaml` statement is present.

**Step 2: Update Pytest Integration Tests**
*   **Action:** In `tests/test_export_live_research_records.py`, add a new test named `test_invalid_yaml_syntax_returns_parse_error`.
*   **Constraint:** Create a temporary file with malformed YAML syntax (e.g., an unclosed bracket or invalid indentation: `"query_groups:\n- [unclosed list\n  key: value"`). Run the export script pointing to this file. Assert that the script returns exit code `1` and that the specific parse error message containing the file path is present in `stderr`.

**Strict Code Guardrails:**
1. Do not remove the existing structural checks for empty/non-dict YAML files; this is purely additive for syntax validation.
2. Run `python -m pytest -q tests/test_export_live_research_records.py` to ensure the new syntax-error test passes alongside the empty/comment-only YAML tests.
3. Automatically execute the **CI-Green Pre-Commit Sequence** (updating `CHANGELOG.txt` and staging files) before finalizing the commit.

***

[the remark this answers is not in the document]
Comment (robertbartlomiejski) [reply scripts/export_live_research_records.py]: @copilot apply changes based on [this feedback](https://github.com/robertbartlomiejski/morskamary/pull/156#discussion_r3232456973)

This is an excellent catch. While your previous updates successfully protected the script from empty files and non-dictionary scalars [1, 2], a syntactically malformed YAML file will still throw a raw `yaml.YAMLError` traceback, creating an opaque crash for the user. 

To fix this gracefully and maintain your established CLI error-handling patterns, we need to wrap the `yaml.safe_load()` call in a `try/except` block and introduce a new test. 

Here is the exact code refactor and the advanced prompt you can give Copilot to execute it:

### 1. Code Implementation to apply
In `scripts/export_live_research_records.py` (around line 268 [1]), the logic should be updated to look like this:

```python
    try:
        query_config = yaml.safe_load(f)
    except yaml.YAMLError as exc:
        print(f"Error: Failed to parse '{path}'. Syntactically invalid YAML:\n{exc}", file=sys.stderr)
        sys.exit(1)
        
    if not isinstance(query_config, dict):
        print(f"Error: '{path}' is empty or not a valid YAML mapping.", file=sys.stderr)
        sys.exit(1)
```

### 2. Advanced Master Prompt for Copilot

You can copy and paste this directly into your GitHub Copilot / Agent terminal to perform the surgical fix and write the tests:

***

**Task Title:** Handle YAML Parse Errors Gracefully in Live Export Script
**Target Repository:** `robertbartlomiejski/morskamary`
**Executive Summary:** `yaml.safe_load()` currently bubbles up a raw `yaml.YAMLError` traceback if the query configuration file is syntactically invalid. Wrap this in a `try/except` block to catch the error, print a clear user-facing message containing the file path, and exit with code 1. This must remain strictly distinct from the structural type validation (`isinstance(query_config, dict)`) implemented previously.

**Step 1: Refactor `scripts/export_live_research_records.py`**
*   **Action:** Locate the `yaml.safe_load(f)` call. 
*   **Constraint:** Wrap it in a `try/except yaml.YAMLError as exc:` block. If caught, print a clear error to `sys.stderr` that includes the file path and the syntax exception details. Call `sys.exit(1)`. Ensure the `import yaml` statement is present.

**Step 2: Update Pytest Integration Tests**
*   **Action:** In `tests/test_export_live_research_records.py`, add a new test named `test_invalid_yaml_syntax_returns_parse_error`.
*   **Constraint:** Create a temporary file with malformed YAML syntax (e.g., an unclosed bracket or invalid indentation: `"query_groups:\n- [unclosed list\n  key: value"`). Run the export script pointing to this file. Assert that the script returns exit code `1` and that the specific parse error message containing the file path is present in `stderr`.

**Strict Code Guardrails:**
1. Do not remove the existing structural checks for empty/non-dict YAML files; this is purely additive for syntax validation.
2. Run `python -m pytest -q tests/test_export_live_research_records.py` to ensure the new syntax-error test passes alongside the empty/comment-only YAML tests.
3. Automatically execute the **CI-Green Pre-Commit Sequence** (updating `CHANGELOG.txt` and staging files) before finalizing the commit.

***

Would you like me to also help you review the `CHANGELOG.txt` entry once Copilot generates it, or should we resume our work on the Stage 2 Provider-Agnostic Metadata Schema?
